src/modules/item/sagas.ts

```diff
@@ -356,14 +356,23 @@ export function* itemSaga(legacyBuilder: LegacyBuilderAPI, builder: BuilderClien
   function* handleSaveItemSuccess(action: SaveItemSuccessAction) {
     const openModals: ModalState = yield select(getOpenModals)
     const location: ReturnType<typeof getLocation> = yield select(getLocation)
+    const { item } = action.payload
     if (openModals['EditItemURNModal']) {
       yield put(closeModal('EditItemURNModal'))
-    } else if (openModals['CreateSingleItemModal'] && location.pathname === locations.collections()) {
+    } else if (openModals['EditPriceAndBeneficiaryModal']) {
+      yield put(closeModal('EditPriceAndBeneficiaryModal'))
+    } else if (openModals['CreateSingleItemModal'] && location.pathname === locations.collections() && item.type === ItemType.WEARABLE) {
       // Redirect to the newly created item detail
-      const { item } = action.payload
       yield put(push(locations.itemDetail(item.id)))
+    } else if (
+      openModals['CreateSingleItemModal'] &&
+      location.pathname === locations.collections() &&
+      item.type === ItemType.EMOTE &&
+      item.beneficiary
```

Comment:
Question: do we need the item to have a `beneficiary` to redirect the `itemEditor`?

Reply:
also, could we add a test for this new behavior? 😃

Reply:
> Question: do we need the item to have a beneficiary to redirect the itemEditor?
This validation avoids redirecting when saving on the first step`(Name, Rarity, Category, Play Mode)` and redirects when the User reaches the last step: `Set a price` where I get the beneficiary(or could be price) to know is the last step.

Could another option be to pass the step in the url and validate with the `pathname`?

> also, could we add a test for this new behavior? 😃
Sure!